tensorflow/contrib/data/python/ops/dataset_ops.py

```diff
@@ -887,6 +887,10 @@ def __init__(self, tensors):
     """See `Dataset.from_tensor_slices()` for details."""
     super(TensorSliceDataset, self).__init__()
     with ops.name_scope("tensors"):
+      tensors = ops.convert_n_to_tensor_or_indexed_slices(tensors)
```

Comment:
You're right, this does seem like a good compromise to force users to use tuples (which actually make more sense in this case).
I'll close the PR.